lib/Controller/ExternalStorageController.php

```diff
@@ -71,21 +71,7 @@ public function getMetadataForSpecificFile(int $fileId): DataResponse {
             }
 
             $fileMetadata = $this->externalStorageService->getMetadataForSpecificFile($user, $fileId);
-
-            if (!isset($fileMetadata['error'])) {
-                if ($fileMetadata['result']['success']) {
-                    return new DataResponse(['success' => true, 'metadata' => $fileMetadata['result']], Http::STATUS_OK);  
-                } else {
-                    return new DataResponse(['success' => false, 'error' => $fileMetadata['result']['error']], Http::STATUS_OK); 
-                } 
-            }
-
-            return new DataResponse([
-                'success' => false, 
-                'error' => $fileMetadata['error'],
-                'errorMessage' => $fileMetadata['result']
-            ], Http::STATUS_INTERNAL_SERVER_ERROR);
-
+            return new DataResponse($fileMetadata, Http::STATUS_OK);
```

Comment:
how are error handled?

Reply:
I return each time a `STATUS_OK` and the function that is called is managed in user a success/error and block catch (line 77 in the ExternalStorageService.php file)

```
if ($response['success']) {
     return ['success' => true, 'metadata' => $response['result']];
} else {
     return ['success' => false, 'error' => $response['error']];
}
```